### PR TITLE
PLT-5905 Fix saml certificates

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Make the certificates for saml consistent

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5905

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has enterprise changes (https://github.com/mattermost/enterprise/pull/123)
- [x] Touches critical sections of the codebase (auth)

### CODE REVIEW

File: api/admin.go [hunk 1]
```diff
@@ -356,12 +356,39 @@ func samlMetadata(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		l4g.Error(err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	m := r.MultipartForm
 
+	certificateFile := ""
+
+	if typeArray, ok := m.Value["type"]; !ok {
```
Comment (jwilander) [inline api/admin.go]: This is a breaking change for this endpoint. Can we make type optional, and if it's not included try to infer the type from the filename extension?

File: api/admin.go [hunk 2]
```diff
@@ -387,7 +414,25 @@ func addCertificate(c *Context, w http.ResponseWriter, r *http.Request) {
 func removeCertificate(c *Context, w http.ResponseWriter, r *http.Request) {
 	props := model.MapFromJson(r.Body)
 
-	if err := app.RemoveSamlFile(props["filename"]); err != nil {
+	certificateFile := ""
+
+	switch certType, _ := strconv.Atoi(props["type"]); certType {
```
Comment (jwilander) [inline api/admin.go]: Same problem here, this is a breaking change. Is there a way we can do this without breaking the endpoint?

**Comment by enahum** [comment]
@jwilander change this to be without braking changes and simple

**Comment by enahum** [comment]
@jwilander unit tests here https://github.com/mattermost/enterprise/pull/123

**Pull request merged**